### ISSUE TITLE
How do I use HrNet bottom-up to get map=64.4% on coco?

### ISSUE THREAD

**Issue by FightStone** [HRNet/Higher-HRNet-Human-Pose-Estimation#2]
First of all, thanks for your outstanding work. 
Secondly, I implemented bottom-up hrnet and highternet according to your paper, I trained both networks using the top-down hrnet's pre-training model, and i set epoch=200, I used the initial learning rate of 0.001, which was reduced by 10 times at 100 and 160 epochs. when epoch=140, map=0.575, It's still a long way from 64.4%! When the learning rate is equal to 0.00001, there is not much room for map gain, so what should i do?
The last, In the issue area of AE code, some said Association Embbeding did not meet the paper's targets. Do you use any tricks?
Looking forward to your reply !

**Comment by panfeng-hover** [comment]
Hi @FightStone, just wondering would you meet the biased foreground and background pixels issue during training for high-resolution feature maps? My network always give me all 0 predictions, and mse loss or binary cross entropy loss not work for me. Thanks!

**Comment by FightStone** [comment]
I don't have that problem and mseloss is suitable for me！The code which i implemented is now only map=61.4% on coco dev datasets. I am trying to improve it,  although it is far away from author's result! :(

**Comment by panfeng-hover** [comment]
Thanks for your reply! Could you help please me take a quick look at my decoder?

```
class Decoder(nn.Module):
    def __init__(self, in_channels, channels, num_elements, init_bias=False):
        super(Decoder, self).__init__()
        self.t_convs = nn.ModuleList()
        self.convs = nn.ModuleList()
        self.init_bias = init_bias
        next_feature = in_channels
        
        for channel in channels:
            self.t_convs.append(
                nn.Sequential(
                    misc_nn_ops.ConvTranspose2d(next_feature, channel, 2, 2, 0),
                    nn.ReLU(inplace=True)
                )
            )

            self.convs.append(
                nn.Sequential(
                    misc_nn_ops.Conv2d(channel, channel // 2, 1, padding=0),
                    misc_nn_ops.Conv2d(channel // 2, num_elements, 1, padding=0)
                )
            )
            
            next_feature = channel + num_elements
        
        self.t_convs.apply(self._init_weights)
        self.convs.apply(self._init_weights)
    
    def _init_weights(self, m):
        if isinstance(m, (misc_nn_ops.Conv2d, misc_nn_ops.ConvTranspose2d)):
            nn.init.kaiming_normal_(m.weight, mode="fan_out", nonlinearity="relu")
            if self.init_bias:
                nn.init.constant_(m.bias, 0)
        
    def forward(self, x):
        outputs = []

        for t_conv, conv in zip(self.t_convs, self.convs):
            x = t_conv(x)
            y = conv(x)
            x = torch.cat((x, y), 1)
            outputs.append(y)
        
        return outputs
```

After I get the `outputs`, I then calculate the loss between all logits heatmaps and ground truth heatmaps, and add them together as the final loss, which somehow not work in my case.

**Comment by FightStone** [comment]
@panfeng-hover Sorry, without context, I don't quite understand. If possible, you can leave a contact information and discuss it together.

**Comment by panfeng-hover** [comment]
@FightStone, thanks for your reply, I found it is due to a bug. I also emailed the author, and told they are planned to release their code by the end of Nov, hope it would help you.

**Comment by leoxiaobin** [comment]
hi, @FightStone @panfeng-hover , we have released our code and models, and feel free to check the code and models.
